src/ts/common/ui/modal.ts

```diff
@@ -37,8 +39,10 @@ const MODAL_BUTTON_CLASS_NAME = "better-library-thing-modal-button";
 const MODAL_INPUT_CLASS_NAME = "better-library-thing-modal-input";
 const MODAL_INPUT_CONTAINER_CLASS_NAME = "better-library-thing-modal-input-container";
 const MODAL_ELEMENT_CONTAINER_CLASS_NAME = "better-library-thing-modal-element-container";
+const MODAL_ID_ATTR = "modal-id";
+const DEFAULT_ID = "_unset";
 
-const activeOverlays = new Set<HTMLElement>();
+const activeModals = new Map<HTMLElement, () => Promise<void>>();
```

Comment:
what is this a map of? modals to exit/dismiss functions? what's the use case for modal specific dismiss functions?

Reply:
This is a map of Modals to dismissals.

The idea is the same as before with `activeOverlays` (which really was `activeModals`):
we don't want to delete loaders.

The modals always had a dismiss function. it just deleted the modal, and the user could decorate it with `onCancel`.

Now we ensure it gets called (both the deleting of the modal and the user provided `onCancel`) when we call `dismissModals`